changes for CentOS 7.1

hoonetorg opened the pull request:
Hi,
as we discussed, would you mind taking a look at these commits (and comments on the commits) and decide if it is something you can merge or if we must discuss and/or rework.
BR
Holger

Commit:
fix typo

Commit:
fix standard owner/mode of files in /etc/one on CentOS 7.1

one controller and sunstone configs are created
user: root
group: oneadmin
mode: 0640
on CentOS 7.1. Hopefully this is the same on all other OS's.
Otherwise this commit makes the formula less compatible.

Commit:
Merge branch 'master' of https://github.com/bechtoldt/saltstack-opennebula-formula

Commit:
make defaults.yaml compatible with CentOS 7.1

make package names os dependent:
  - CentOS 7.1 packages have different names
  - rubygem package names  are added to controller
    (this is WIP  because I do not know which rubygems are needed
    for what - controller, sunstone, gate or flow)
  - optional groups for user oneadmin are probably debian specific.
    That is why they are now OS dependent
    (I only needed to make "usermod -a -G oneadmin qemu" on
    compute nodes)
  - Per default only one_key and one_auth are now managed. This
    sets only the credentials for the oneadmin user.

    It is not possible to manage ec2_auth, occi_auth, oneflow_auth,
    onegate_auth or sunstone_auth on the 1st start of opennebula.
    That is why these settings are commented out now.

    When oned starts the first time, the database will be
    bootstrapped. If one of the _auth files in /var/lib/one/.one
    exists, oned complains:
    "Password file /var/lib/one//.one/<name>_auth already exists
    but OpenNebula is boostraping the database.
    Check your database configuration in oned.conf."

    This is still a WIP, because as far as I understand,
    it's not possible to simply create or
    change the _auth-files in /var/lib/one/.one to update the
    credentials, you also need to update the password either with
    the "oneuser" command or directly in the database(if you know,
    what you are doing).

    At this stage of the formula, simply updating the one_key and
    one_auth in pillar data would fail without changing the password
    in the database/with oneuser command.

arnisoph commented on opennebula/controller.sls:
```diff
@@ -57,9 +57,9 @@ one_controller_config_{{ c }}:
     - contents_pillar: opennebula:lookup:controller:config:{{ c }}:contents
   {% endif %}
-    - user: {{ f.user|default('oneadmin') }}
+    - user: {{ f.user|default('root') }}
```
could you please explain the reason for this change?

hoonetorg replied:
an ls -laR on /var/lib/one shows that per default (under CentOS 7.1) most files are created with owner oneadmin:oneadmin and permission permission 0644 in this folder. 
So if it is the same on all platforms, it's probably good to use these settings per default.
To be honest, I have no Debian, Ubuntu, SuSe, Arch, Gentoo available - and no time - to hv a look at this.

arnisoph commented on opennebula/defaults.yaml:
```diff
@@ -2,50 +2,42 @@ kernel:
         manage:
-          - ec2_auth
-          - occi_auth
+#          - ec2_auth
```
you can override data that is listed in the defaults.yaml, please keep these comments only if we really need them:

```
opennebula:
  lookup:
    oneadmin:
      config:
        manage:
          - one_auth
          - one_key
...
```

hoonetorg replied:
Yeah..., I' ve seen and tried this, settings in the pillar override default settings:
so doing a:
opennebula:
  lookup:
    oneadmin:
      config:
        manage: []

prevents configuring this files at all.
But still I'm a bit scared of changing the auth files on every salt run.
For one_auth and one_key I would like to see something like this
file.managed:
  ...
  - unless test -f /var/lib/one/.one/one_auth
  ...
because simply changing the file without changing the database does not work.
I will implement a patch where I create those 2 files if they don't exist.
I will send you a pull request for this and we can discuss again if you like it :)

hoonetorg replied:
OK, last note is already so long, but I there is still something to say:
Managing
- ec2_auth
- occi_auth
- oneflow_auth
- onegate_auth
- sunstone_auth
on initial run will not work (I described it in the comment of the commit), because oned complains about existing _auth-files before the database is bootstrapped.
Therefor I suggest disabling managing these files in the default.yaml, because installation will fail otherwise.
If somebody is knowing what he's doing he can override it in the pillar data.

arnisoph commented on opennebula/onegate.sls:
```diff
@@ -3,7 +3,7 @@
 {% set service = datamap.onegate.service|default({}) %}
 
-include: {{ salt['pillar.get']('opennebula:lookup:onegate:sls_include', ['.contoller']) }}
+include: {{ salt['pillar.get']('opennebula:lookup:onegate:sls_include', ['.controller']) }}
```
good catch!

Commit:
completely disable managing _auth files per default.
It will fail on initial run of formula (at least on controller nodes).

Creating an /var/lib/one/.one/one_key and /var/lib/one/.one/one_auth
if they don't exist is perfectly fine and wanted, but will be
implemented in another commit.

Commit:
make last commit complete

it's perfectly fine to define metadata for oneadmin _auth-files in
default.yaml, as long as they are not managed.
